include/solvers/unsteady_solver.h

```diff
@@ -115,13 +117,28 @@ class UnsteadySolver : public TimeSolver
    */
   virtual void retrieve_timestep () override;
 
+  /**
+   * A method to integrate the system::QoI functionals.
+   */
+  virtual void integrate_qoi_timestep() override;
+
   /**
    * A method to integrate the adjoint sensitivity w.r.t a given parameter
    * vector. int_{tstep_start}^{tstep_end} dQ/dp dt = int_{tstep_start}^{tstep_end} (\partialQ / \partial p) - ( \partial R (u,z) / \partial p ) dt
-   * The midpoint rule is used to numerically integrate the timestep
+   * The trapezoidal rule is used to numerically integrate the timestep.
    */
   virtual void integrate_adjoint_sensitivity(const QoISet & qois, const ParameterVector & parameter_vector, SensitivityData & sensitivities) override;
 
+  /**
+   * A method to compute the adjoint refinement error estimate at the current timestep.
+   * int_{tstep_start}^{tstep_end} R(u^h,z) dt
+   * Fills in an ErrorVector that contains the weighted sum of errors from all the QoIs and can be used to guide AMR.
+   * Also fills in a map that links QoI indices to spatially integrated error estimates for the QoI with that index.
+   * The trapezoidal rule (for temporally smooth QoI) or left Riemann sum (for an instantaneous QoI) is used
+   * for the numerical integration depending on the nature of the QoI.
```

Comment:
Let's either just copy and paste the time_solver.h comment into its subclasses, or have any differences be due to subclass-specific underlying differences?  E.g., the documentation for QoI_time_instant is missing, so I can't quickly see it when looking at this class, yet there's so much documentation here that neither can I conclude that the real documentation must be elsewhere.  It's like antimissile chaff for doxygen users!

Reply:
Done.